File: app/src/main/java/com/dgd/earthquakes/BaseApplication.java
package com.dgd.earthquakes;

import android.app.Application;

import com.dgd.earthquakes.injection.ComponentProvider;
import com.dgd.earthquakes.injection.Injection;

/**
 * Created by Max
 * on 30-Apr-17.
 */
public class BaseApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        Injection.setComponentProvider(new ComponentProvider(this));
    }
}

File: app/src/main/java/com/dgd/earthquakes/BaseApplication.kt
package com.dgd.earthquakes

import android.app.Application

import com.dgd.earthquakes.data.IRepo

/**
 * Created by Max
 * on 30-Apr-17.
 */

class BaseApplication : Application() {
    var repository: IRepo? = null

    override fun onCreate() {
        super.onCreate()
        instance = this
        repository = Injection.repository
    }

    companion object {
        var instance: BaseApplication? = null
            private set
    }
}
